### PR TITLE
docs: use templates to reuse documentation properties

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs-src/api-params.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  - `'domcontentloaded'` - consider operation to be finished when the `DOMContentLoaded` event is fired.
+  - `'load'` - consider operation to be finished when the `load` event is fired.
+  - `'networkidle'` - consider operation to be finished when there are no network connections for at least `500` ms.
+
+## navigation-timeout
+- `timeout` <[number]> Maximum operation time in milliseconds, defaults to 30 seconds, pass `0` to disable timeout. The default value can be changed by using the [browserContext.setDefaultNavigationTimeout(timeout)](#browsercontextsetdefaultnavigationtimeouttimeout), [browserContext.setDefaultTimeout(timeout)](#browsercontextsetdefaulttimeouttimeout), [page.setDefaultNavigationTimeout(timeout)](#pagesetdefaultnavigationtimeouttimeout) or [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) methods.
+
+## wait-for-timeout
+- `timeout` <[number]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [browserContext.setDefaultTimeout(timeout)](#browsercontextsetdefaulttimeouttimeout).
```
Comment (dgozman) [inline docs-src/api-params.md:11]: Hmm... Why doesn't this mention page.setDefaultTimeout? Perhaps this is different for page.waitForX and context.waitForX?

File: docs-src/api-params.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+## input-button
+- `button` <"left"|"right"|"middle"> Defaults to `left`.
+
+## input-files
+- `files` <[string]|[Array]<[string]>|[Object]|[Array]<[Object]>>
+  - `name` <[string]> [File] name **required**
+  - `mimeType` <[string]> [File] type **required**
+  - `buffer` <[Buffer]> File content **required**
+
+## input-down-up-delay
```
Comment (dgozman) [inline docs-src/api-params.md:42]: Are we after full reuse? This one, for example, looks very minor, perhaps we can copy it?

File: docs/api.md [hunk 3]
```diff
@@ -299,23 +299,23 @@ Creates a new browser context. It won't share cookies/cache with other browser c
...
   - `videoSize` <[Object]> **NOTE** Use `recordVideo` instead, it takes precedence over `videoSize`. Specifies dimensions of the automatically recorded video. Can only be used if `videosPath` is set. If not specified the size will be equal to `viewport`. If `viewport` is not configured explicitly the video size defaults to 1280x720. Actual picture of the page will be scaled down if necessary to fit specified size.
     - `width` <[number]> Video frame width.
     - `height` <[number]> Video frame height.
-  - `recordHar` <[Object]> Enables [HAR](http://www.softwareishard.com/blog/har-12-spec) recording for all pages into `recordHar.path` file. If not specified, the HAR is not recorded. Make sure to await [`page.close`](#pagecloseoptions) for the HAR to be saved.
+  - `recordHar` <[Object]> Enables [HAR](http://www.softwareishard.com/blog/har-12-spec) recording for all pages into `recordHar.path` file. If not specified, the HAR is not recorded. Make sure to await [`browserContext.close`](#browsercontextclose) for the HAR to be saved.
```
Comment (dgozman) [inline docs/api.md:306]: This is a minor regression.

File: docs/api.md [hunk 4]
```diff
@@ -2030,7 +2030,7 @@ Video object associated with this page.
 - `event` <[string]> Event name, same one would pass into `page.on(event)`.
 - `optionsOrPredicate` <[Function]|[Object]> Either a predicate that receives an event or an options object.
   - `predicate` <[Function]> receives the event data and resolves to truthy value when the waiting should resolve.
-  - `timeout` <[number]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [browserContext.setDefaultTimeout(timeout)](#browsercontextsetdefaulttimeouttimeout) or [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) methods.
+  - `timeout` <[number]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [browserContext.setDefaultTimeout(timeout)](#browsercontextsetdefaulttimeouttimeout).
```
Comment (dgozman) [inline docs/api.md:2033]: This is a regression.

File: docs/api.md [hunk 5]
```diff
@@ -2041,7 +2041,7 @@ is fired.
 - `arg` <[EvaluationArgument]> Optional argument to pass to `pageFunction`
 - `options` <[Object]> Optional waiting parameters
   - `polling` <[number]|"raf"> If `polling` is `'raf'`, then `pageFunction` is constantly executed in `requestAnimationFrame` callback. If `polling` is a number, then it is treated as an interval in milliseconds at which the function would be executed. Defaults to `raf`.
-  - `timeout` <[number]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) method.
+  - `timeout` <[number]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [browserContext.setDefaultTimeout(timeout)](#browsercontextsetdefaulttimeouttimeout).
```
Comment (dgozman) [inline docs/api.md:2044]: This was wrong, and now it's also wrong, but differently 😄

File: docs/api.md [hunk 6]
```diff
@@ -4567,12 +4566,13 @@ const browser = await chromium.launch({  // Or 'firefox' or 'webkit'.
...
   - `videoSize` <[Object]> **NOTE** Use `recordVideo` instead, it takes precedence over `videoSize`. Specifies dimensions of the automatically recorded video. Can only be used if `videosPath` is set. If not specified the size will be equal to `viewport`. If `viewport` is not configured explicitly the video size defaults to 1280x720. Actual picture of the page will be scaled down if necessary to fit specified size.
     - `width` <[number]> Video frame width.
     - `height` <[number]> Video frame height.
-  - `recordHar` <[Object]> Enables [HAR](http://www.softwareishard.com/blog/har-12-spec) recording for all the pages into `recordHar.path` file. If not specified, HAR is not recorded. Make sure to await [`page.close`](#pagecontext) for HAR to be saved.
-    - `omitContent` <[boolean]> Optional setting to control whether to omit request content from the HAR. Defaults to false.
+  - `recordHar` <[Object]> Enables [HAR](http://www.softwareishard.com/blog/har-12-spec) recording for all pages into `recordHar.path` file. If not specified, the HAR is not recorded. Make sure to await [`browserContext.close`](#browsercontextclose) for the HAR to be saved.
```
Comment (dgozman) [inline docs/api.md:4574]: This is a bug fix!

**Review by dgozman — APPROVED** [review]
Looks great, with some minor issues!

File: docs-src/api-params.md (hunk 2 above)
Comment (pavelfeldman) [reply docs-src/api-params.md:42]: 8 occurances

File: docs-src/api-params.md (hunk 1 above)
Comment (pavelfeldman) [reply docs-src/api-params.md]: That's what it was!

File: docs/api.md (hunk 3 above)
Comment (pavelfeldman) [reply docs/api.md:306]: Yeah, but I think it is Okay

File: docs/api.md (hunk 4 above)
Comment (pavelfeldman) [reply docs/api.md:2033]: I think using context's one is better...

File: docs/api.md (hunk 5 above)
Comment (pavelfeldman) [reply docs/api.md:2044]: I think using context's one is better...

**Pull request merged**